### PR TITLE
fix: checkbox invalid message to match figma kit

### LINKED ISSUE #18033: [Bug]: invalidText of horizontal CheckboxGroup is rendered on wrong place
### Package

@carbon/react

### Browser

Firefox

### Package version

1.17.0

### React version

18.3.0

### Description

`CheckboxGroup` component renders the `invalidText` at the different position from the figma library.
With `orientation="horizontal"`, `invalidText` shows at the right of the `Checkbox`s, and I expected it to be shown at the bottom.

Here is the screenshot from our Figma sketch:
Image

And here's the screenshot of the example react component:
Image

### Reproduction/example

https://stackblitz.com/edit/github-emxaww?file=src%2FApp.jsx

### Steps to reproduce

1. Create figma checkbox group with Horizontal=on, and state=Invalid
2. Compare the rendered component with the provided stackblitz example

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Application/PAL

_No response_

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by riddhybansal**
Closes #18033 

CheckboxGroup component renders the invalidText at the different position from the figma library.
With orientation="horizontal", invalidText shows at the right of the Checkboxs, and I expected it to be shown at the bottom.

#### Changelog

**Changed**

- Seperated the container of error message from the flex container of radio buttons and hence modified the classes

#### Testing / Reviewing

Go to checkbox and check if invalid message is rendered correctly at bottom

### CODE REVIEW

**Review by guidari — CHANGES_REQUESTED** [review]
Hey Riddhi, It is looking good! Just one thing.
I think we should keep the `invalid` and `readonly` message inside the `fieldset`. If you test with VO in the [current Storybook](https://react.carbondesignsystem.com/?path=/story/components-checkbox--playground&args=invalid:!true), you’ll notice that they are grouped together, which helps the screen reader user understand that the message is part of the same group label. 
We can also confirm that with Raghu if needed.

**Comment by riddhybansal** [comment]
> Hey Riddhi, It is looking good! Just one thing. I think we should keep the `invalid` and `readonly` message inside the `fieldset`. If you test with VO in the [current Storybook](https://react.carbondesignsystem.com/?path=/story/components-checkbox--playground&args=invalid:!true), you’ll notice that they are grouped together, which helps the screen reader user understand that the message is part of the same group label. We can also confirm that with Raghu if needed.

Oh I see what you are saying !!

**Comment by riddhybansal** [comment]
Hey @guidari Look at this found an alternative way to fix that bug . Lemme know what do you think about it ?

File: packages/react/src/components/CheckboxGroup/CheckboxGroup.tsx [hunk 1]
```diff
@@ -65,13 +65,16 @@ const CheckboxGroup: React.FC<CheckboxGroupProps> = ({
...
     [`${prefix}--checkbox-group--readonly`]: readOnly,
     [`${prefix}--checkbox-group--invalid`]: !readOnly && invalid,
     [`${prefix}--checkbox-group--warning`]: showWarning,
     [`${prefix}--checkbox-group--slug`]: slug,
   });
 
+  const orientationClass =
+    orientation === 'horizontal'
+      ? `${prefix}--checkbox-group--${orientation}`
+      : '';
```
Comment (guidari) [inline packages/react/src/components/CheckboxGroup/CheckboxGroup.tsx]:
```suggestion
      : undefined;
```
LGTM! Let's just change to `undefined` to hide the empty class.

Screenshot 2024-11-18 at 10 38 00

**Review by guidari — APPROVED** [review]
LGTM! 🚀

**Review by thyhmdo — APPROVED** [review]

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
This does technically fix the bug, however I think it would be preferable if we just update the styles vs. changing the underlying html structure. I believe you should be able to add a  inline-size (width) of 100% to the invalid or warn text when its horizontal and it will display correctly below the checkboxes.

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**